src/vtt-segment-loader.js

```diff
@@ -976,6 +980,22 @@ export default class VTTSegmentLoader extends videojs.EventTarget {
 
     segment.requested = true;
 
+    // prepend the media initialization segment if it exists
+    if (segment.map) {
+      const initId = initSegmentId(segment.map);
+      const initSegment = this.initSegments_[initId];
+
+      segment.map.bytes = initSegment.bytes;
+
+      const vttHeader = new Uint8Array('WEBVTT\n\n'.split('').map(char => char.charCodeAt(0)));
+      const combinedByteLength = vttHeader.byteLength + segmentInfo.bytes.byteLength;
+      const combinedSegment = new Uint8Array(combinedByteLength);
+
+      combinedSegment.set(vttHeader);
+      combinedSegment.set(segmentInfo.bytes, vttHeader.byteLength);
+      segmentInfo.bytes = combinedSegment;
```

Comment:
So the webvtt parser is very strict and there must be a blank line after the header. I just checked and you're right we dont need the entire combined here, but we do at least need a new line. (in the source i am testing, the init segment contains a newline at the end, but my guess is most common would be that the init segment contains no new lines at the end). So we do need some form of combined here so that the parser works properly. I think that leaves us with 3 options

1. leave it how it is here
2. prepend 2 newline characters to the segment bytes
3. append 2 newline character to the end of init segment bytes